File: app/src/main/java/com/work/ddnet/rx/RxRestClientBuilder.java
package com.work.ddnet.rx;

import android.content.Context;



import java.io.File;
import java.util.WeakHashMap;

import okhttp3.MediaType;
import okhttp3.RequestBody;

/**
 * @author Ting
 * @date 2017/11/17
 */

public class RxRestClientBuilder {

    private static final WeakHashMap<String, Object> PARAMS = RestCreator.getParams();
    private String mUrl = null;
    private RequestBody mBody = null;
    private File mFile = null;


    public final RxRestClientBuilder url(String url) {
        this.mUrl = url;
        return this;
    }

    public final RxRestClientBuilder params(WeakHashMap<String, Object> params) {
        PARAMS.putAll(params);
        return this;
    }

    public final RxRestClientBuilder params(String key, Object value) {
        PARAMS.put(key, value);
        return this;
    }

    public final RxRestClientBuilder raw(String raw) {
        this.mBody = RequestBody.create(MediaType.parse("application/json;charset=UTF-8"), raw);
        return this;
    }

    public final RxRestClientBuilder file(File file) {
        this.mFile = file;
        return this;
    }

    public final RxRestClientBuilder file(String file) {
        this.mFile = new File(file);
        return this;
    }

    public final RxRestClient build() {
        return new RxRestClient(mUrl, PARAMS, mBody, mFile);
    }
}

File: app/src/main/java/com/work/ddnet/rx/RxRestClientBuilder.kt
package com.work.ddnet.rx


import java.io.File
import java.util.WeakHashMap

import okhttp3.MediaType
import okhttp3.RequestBody

/**
 * @author Ting
 * @date 2017/11/17
 */

class RxRestClientBuilder {
    private var mUrl: String = ""
    private var mBody: RequestBody? = null
    private var mFile: File? = null


    fun url(url: String): RxRestClientBuilder {
        this.mUrl = url
        return this
    }

    fun params(params: WeakHashMap<String, Any>): RxRestClientBuilder {
        PARAMS.putAll(params)
        return this
    }

    fun params(key: String, value: Any): RxRestClientBuilder {
        PARAMS.put(key, value)
        return this
    }

    fun raw(raw: String): RxRestClientBuilder {
        this.mBody = RequestBody.create(MediaType.parse("application/json;charset=UTF-8"), raw)
        return this
    }

    fun file(file: File): RxRestClientBuilder {
        this.mFile = file
        return this
    }

    fun file(file: String): RxRestClientBuilder {
        this.mFile = File(file)
        return this
    }

    fun build(): RxRestClient {
        return RxRestClient(mUrl, PARAMS, mBody, mFile)
    }

    companion object {
        private val PARAMS = RestCreator.getParam()
    }
}
